### ISSUE TITLE
[Offline Posting] Support published, private, and pending statuses

### ISSUE THREAD

**Issue by shiki** [wordpress-mobile/WordPress-iOS#12587]
📣 Part of #12227. Please target the `issue/12240-master-branch` branch. 
⚠️ Kinda blocked by the copy review of https://github.com/wordpress-mobile/WordPress-Android/pull/10542 but can still be worked on. 

At the time of this writing, all the auto-upload/auto-save/cancellation flow completed in `issue/12240-master-branch` only supports posts whose `status` (`BasePost.Status`) are:

- `.draft`
- `.publish`

## Goal

The goal is to support:

- `.publishPrivate`
- `.pending`
- `.scheduled`

These types are ignored and are out of scope:

- `.trash`
- `.deleted`

## Messages

Most of the work that will be done are adding messages. Please make sure that we are using the copy-reviewed messages as described in https://github.com/wordpress-mobile/WordPress-Android/pull/10542.

**Comment by shiki** [comment]
Done in #12613
